internal/graph/graph.go

```diff
@@ -41,13 +42,32 @@ func ResolutionDepthFromContext(ctx context.Context) (uint32, bool) {
 	return depth, ok
 }
 
-type ResolutionMetadata struct {
+type CheckRequestMetadata struct {
```

Comment:
```suggestion
type ResolveCheckRequestMetadata struct {
```

This to distinguish between the parent Check RPC. We should strive to stay consistent with the internal dispatch RPC signature, which is `ResovleCheck`.